website_twitter_no_ext_links/models/twitter_feed.py

```diff
@@ -0,0 +1,147 @@
+        comodel_name='twitter.theme',
+        help="Default Theme, if the user does not select it from the widget it is the fallback for this feed",
+        required=True
+    )
+                    
+
+    def get_token(self, c_key, c_secret):
+
+        url = 'https://api.twitter.com/oauth2/token'
+        credentials = base64.b64encode(
+                '%s:%s' % (
+                    urllib.quote(c_key),
+                    urllib.quote(c_secret),
+                ))
+        request = urllib2.Request(url, data='grant_type=client_credentials')
+        request.add_header("Authorization", "Basic %s" % credentials)
+        request.add_header(
+            "Content-Type",
+            "application/x-www-form-urlencoded;charset=UTF-8")
+        response = urllib2.urlopen(request)
+        response_str = response.read()
+        response_dict = json.loads(response_str)
+        return str(response_dict['access_token'])
+
+    @api.model
+    def get_all_twitter_feeds(self):
+        feeds = self.env['twitter.feed'].search([])
+        for feed in feeds:
+            token = self.get_token(
+                feed.twitter_api_key,
+                feed.twitter_api_secret
+            )
+            args = {
+                    'screen_name': feed.twitter_screen_name,
+                    'count': feed.feed_length,
+                    }
+            url = feed.base_twitter_api_path + "statuses/user_timeline.json?%s" % urllib.urlencode(
+                args)
+            request = urllib2.Request(url)
+            request.add_header(
+              "Authorization",  "Bearer %s" %  token
+            )
+            request.add_header(
+                'content-type',  'application/x-www-form-urlencoded;charset=UTF-8'
+            )
+            try:
+                response = urllib2.urlopen(request)
+            except urllib2.HTTPError as e:
+                print e.code
+                print e.reason
+                print e.read()
+            userposts = response.read()
+            userposts_dict = json.loads(userposts)
+            if userposts_dict:
+                feed.clear_tweets()
+                for item in userposts_dict:
+                    tweet_formatted_text = item['text']
+                    at_users = re.findall('(?<=@)\w+', tweet_formatted_text)
+                    hashtags = re.findall('(?<=#)\w+', tweet_formatted_text)
+                    for at_user in at_users:
+                        tweet_formatted_text = tweet_formatted_text.replace(
+                            '@'+at_user,
+                            '<span color="' + feed.theme.twitter_at_color + '">@' + at_user + '</span>'
+                        )
+                    for hashtag in hashtags:
+                        tweet_formatted_text = tweet_formatted_text.replace(
+                            '#'+hashtag,
+                            '<span color="' + feed.theme.twitter_hashtag_color + '">#' + hashtag + '</span>'
+                        )
```

Comment:
I think you'll be better off storing the tweets unchanged, and add this on display. Why? Because then we can change the display according to the platform, and when we add other stuff afterwards, it also applies to old tweets